### PR TITLE
Minor improvement for SparseFileTracker toString

### PR DESCRIPTION
**Pull request by ywangd**
Include both length and complete in toString for easier introspection. Also add a comment to the choice of using a string constant (`file`) for the description.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java [hunk 1]
```diff
@@ -715,7 +715,7 @@ class CacheFileRegion extends EvictableRefCounted {
         CacheFileRegion(RegionKey<KeyType> regionKey, int regionSize) {
             this.regionKey = regionKey;
             assert regionSize > 0;
-            tracker = new SparseFileTracker("file", regionSize);
+            tracker = new SparseFileTracker(regionKey.toString(), regionSize);
```
Comment (tlrx) [inline x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java:718]: I think we used `file` here to avoid consuming a lot of heap for the description. I would prefer to revert this change, unless we have a good reason to do it?

Comment (ywangd) [reply x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java]: Thanks for explaining the subtlety. I have reverted this change and also added a comment about the choice of using string constant. 

Btw, we have other places such as the following that uses non-constant string. Is it a concern or not?

https://github.com/elastic/elasticsearch/blob/eeecdbf87b08ec03682dabe7def7272055119252/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java#L137

Comment (tlrx) [reply x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java]: Thanks. 

I haven't seen Cold nodes heap dumps with many heap consumed by those descriptions but it looks like we could save a bit of heap by reducing the description there too.

**Review by tlrx — APPROVED** [review]
LGTM

File: x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java (hunk 1 above)
Comment (ywangd) [reply x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java]: I'll leave them for now since there has not been any issue to avoid any surprises. Thanks!
